### ISSUE TITLE
Query on Accessibility Insights Potentially Flagging Issues From Other Pages

### ISSUE THREAD

**Issue by daphneyyt** [microsoft/accessibility-insights-for-android-service#77]
Hi, we would like to make a few queries on WCAG potentially flagging issues on other pages of the mobile app, when the elements in question are not present on those pages.

In this post, I am asking 3 overarching questions (labelled as 'Examples', as I am sharing the context before asking the follow-up sub-questions). Hence this post might be a little long. Please bear with us :)

Thank you in advance and please feel free to reach out if further information is required. However, we won't be able to share the app details as the app requires login details that are confidential.

### **Example 1**
When scanning Mobile App A for WCAG 1.3.1, WCAG 3.3.2, the elements raised in the following screens are shown in step 1 of the form submission.

<img width="960" alt="App A - 131 - 1b" src="https://user-images.githubusercontent.com/75086091/108952075-fa899700-76a3-11eb-9f1f-2aeecba7c329.png">

We believe these are the elements flagged by AIFA as issues:
<img width="960" alt="App A - 131 - 1c" src="https://user-images.githubusercontent.com/75086091/108952163-05442c00-76a4-11eb-9eaf-cebd4a50520b.png">

However, on the confirmation screen on step 2, these elements are not present. Yet they are flagged by  Accessibility Insights as issues
<img width="960" alt="App A - 131 - 2" src="https://user-images.githubusercontent.com/75086091/108952194-10975780-76a4-11eb-8f72-9a01fbfafc74.PNG">

This is how the screen looks like without the visual helper
<img width="1044" alt="App A - 131 - 2b" src="https://user-images.githubusercontent.com/75086091/108952221-1a20bf80-76a4-11eb-88ff-8dbe31108496.png">

### **Questions For Example 1**
1)	May we know what is causing Accessibility Insights to flag these WCAG issues on Step 2 of the mobile app when the flagged elements are not present?
2)	Do these ‘flagged’ issues raise any potential accessibility issues? Or can these be ignored?
3)	App owner claims that assistive technologies are already in place for Step 2 pages. Is there any way we could validate this? What does Microsoft recommend?

##
### **Example 2**
The following are errors flagged when we scan the following pages for WCAG. For context, this is a page where the mobile app is supposed to calculate wages contribution for the user after the user keys in the required information.

These are the issues flagged in Step 1 of 2
<img width="960" alt="App A - SC - AIFA" src="https://user-images.githubusercontent.com/75086091/108951594-36702c80-76a3-11eb-9adf-7d6d288440e7.png">

The same issues are flagged in Step 2 of 2. However, the elements are not present on this page
<img width="960" alt="App A - SC - AIFA Step 2a" src="https://user-images.githubusercontent.com/75086091/108951681-6c151580-76a3-11eb-985b-b323e6ebf32d.png">
<img width="960" alt="App A - SC - AIFA Step 2b" src="https://user-images.githubusercontent.com/75086091/108951689-6f100600-76a3-11eb-82ce-1ae0aa12bc39.png">

### **Questions For Example 2**
1)	May we know what is causing Accessibility Insights to flag these WCAG issues on Step 2 of the mobile app when the elements are not present?
2)	Do these ‘flagged’ issues raise any potential accessibility issues? Or can these be ignored?
3)	App owner claims that assistive technologies are already in place for Step 2 pages. Is there any way we could validate this? What does Microsoft recommend?

##
### **Example 3**
This is a page with 2 tabs: Citizen and Permanent Resident, meaning, they share the same page on the mobile app. 

These are the errors flagged when we scanned the Citizen page:
<img width="960" alt="App A - SC - AIFA" src="https://user-images.githubusercontent.com/75086091/108951899-be563680-76a3-11eb-96e2-2a1b52731c89.png">

These are the errors flagged when we scan the Permanent Resident page. As you can see, the errors on the Permanent Resident tab is the same as the Citizen’s tab

if the entries are filled on the Citizen’s tab, the information will be ported over when we select the Permanent Resident’s tab
<img width="959" alt="App A - PR - AIFA" src="https://user-images.githubusercontent.com/75086091/108951976-d928ab00-76a3-11eb-9b01-fc8d2b6882cc.png">

### **Questions For Example 3**
1)	In such instances, could it be possible that the elements flagged by Accessibility Insights are the same between the 2 tabs – and if the app owner rectifies the source issue, the accessibility issue should be resolved on both tabs of the mobile app? Or;
2)	Should we advise the app owner to treat each tab’s finding as separate entities and rectify the codes accordingly?

**Comment by lisli1** [comment]
Hello @daphneyyt, thank you for using Accessibility Insights. 

Given the information that I can gather from the screenshots, my guess is that the elements are being hidden but are still present/active on the screen, which would show up to assistive technologies and thus cause the components to be flagged as issues, even if they are not visible.

Examples 1 & 2:

1. It depends on how the elements are displayed and switched out of view; can't tell from screenshots.
2. Again, I can't give a definitive answer on whether the issues can be ignored, based solely on the screenshots.
3. I'm not sure what you mean by this question, particularly "assistive technologies are already in place." Can you clarify?

Example 3:

1. It is possible that the elements are the same, but it's hard to tell from the screenshots. You can try to fix the issue on one tab and see if the issues are still being flagged for the other tab when you re-run.
2. Accessibility Insights for Android will flag all failure instances, so it is best to fix the app so that the issues are no longer flagged. The actual fix will depend on how the components in the app are programmed.

Ultimately, based on the information shown in the screenshots, it's a bit difficult to determine what the exact issues are; I can only speculate what _might_ be the issue, but the only way to actually diagnose the problem is if we can get a working repro of the issues. 

**Is there any way you can provide us with a minimal version of the app with just the components in question?**
